common/testing/testvars/test_vars.go

```diff
@@ -104,38 +90,74 @@ func (tv *TestVars) clone() *TestVars {
 
 	return tv2
 }
+func (tv *TestVars) cloneSetVal(key string, val any) *TestVars {
+	tv2 := tv.clone()
+	tv2.kv.Store(key, val)
+	return tv2
+}
 
-func (tv *TestVars) cloneSet(typ string, key []string, val any) *TestVars {
-	newTv := tv.clone()
-	newTv.set(typ, key, val)
-	return newTv
+func (tv *TestVars) cloneAppendString(key string, initialValGen func(key string) string, suffix string) *TestVars {
+	tv2 := tv.clone()
+
+	v, isLoaded := tv.kv.Load(key)
+	if !isLoaded {
+		v = initialValGen(key)
+	}
+
+	vString, vIsString := v.(string)
+	if !vIsString {
+		vStringer, vIsStringer := v.(fmt.Stringer)
+		if !vIsStringer {
+			panic(fmt.Sprintf("value of key %s is of type %T but must be of type %T or implement fmt.Stringer", key, v, ""))
+		}
+		vString = vStringer.String()
+	}
+	tv2.kv.Store(key, vString+"_"+suffix)
+
+	return tv2
 }
 
-// ----------- Methods for every "type" ------------
-// TODO: add more as you need them.
+// ----------- Methods for every entity ------------
```

Comment:
I think this all becoming more complicated.

I recommend to restrict this to "With" approach. And remove those "AppendTo" functions.
then users of this code will use just
`tv = tv.WithWorkflowID("specific_id")`
this will create custom "workflow_id" entry with "tv.uniqueString+"workflow-id_"+specific_id.

Reply:
I also will recommend to add "At" function:
`func (tv *TestVars) At(key string, initialVal...any)`
which will be basically return getOrCreate(tv.uniqueString, initialVal)